### PR TITLE
Changes to allow update to arkworks

### PR DESCRIPTION
**Pull request by mimoo**
This PR allows us to upgrade the Marlin repository, so that we can further down move from this fork to arkworks.

Note that it reverts the last commit, which didn't build on any version of Rust I tried (1.45.2 and 1.51.1)

### CODE REVIEW

File: algebra-core/src/curves/models/short_weierstrass_jacobian.rs [hunk 1]
```diff
@@ -603,19 +603,19 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
...
             // C = B^2
-            let c = b.square();
+            let mut c = b.square();
 
             // D = 2*((X1+B)2-A-C)
             let d = ((self.x + &b).square() - &a - &c).double();
 
             // E = 3*A
-            let e = a + a.double();
+            let e = a + a.double_in_place();
```
Comment (mrmr1993) [inline algebra-core/src/curves/models/short_weierstrass_jacobian.rs:618]: Why have we switched to `double_in_place` here?

**Review by mrmr1993 — COMMENTED** [review]
Looks good except the `double_in_place` thing.

File: algebra-core/src/curves/models/short_weierstrass_jacobian.rs (hunk 1 above)
Comment (mimoo) [reply algebra-core/src/curves/models/short_weierstrass_jacobian.rs:618]: that's reverting vitaly's commit, we've always used double_in_place

Comment (mimoo) [reply algebra-core/src/curves/models/short_weierstrass_jacobian.rs:618]: (see https://github.com/o1-labs/zexe/blob/21a9eb6948083dcad7aa21c61da6feac05e28319/algebra-core/src/curves/models/short_weierstrass_jacobian.rs#L618)

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**